### PR TITLE
Enable adding cap insets

### PR DESCRIPTION
**Pull request by neelmistry94**
## Pull request checklist:
 - [x] Briefly describe the changes in this PR.
 - [x] Apply changelog label ('breaking change', 'bug :beetle:', 'build', 'docs', 'feature :green_apple:', 'performance :zap:', 'testing :100:') or use the label 'skip changelog'

### Summary of changes
This PR removes default parameters from the `addImage` function so that it will not be a breaking change post GA.

### CODE REVIEW

File: Sources/MapboxMaps/Location/LocationStyleProtocol.swift [hunk 1]
```diff
@@ -8,15 +8,30 @@ internal protocol LocationStyleProtocol: AnyObject {
...
+                           sdf: Bool,
+                           stretchX: [ImageStretches],
+                           stretchY: [ImageStretches],
+                           content: ImageContent?) throws {
+        try addImage(image,
+                     id: id,
+                     sdf: sdf,
+                     stretchX: stretchX,
+                     stretchY: stretchY,
+                     content: content)
```
Comment (macdrevx) [inline Sources/MapboxMaps/Location/LocationStyleProtocol.swift:35]: Trying to understand how this isn't a recursive call

Comment (macdrevx) [reply Sources/MapboxMaps/Location/LocationStyleProtocol.swift:35]: This is in a protocol extension which must have something to do with it. Let's see if we can find a way not to do this.

File: Sources/MapboxMaps/Style/Style.swift [hunk 2]
```diff
@@ -471,7 +471,12 @@ extension Style: StyleManagerProtocol {
 
     // MARK: - Style images
 
-    public func addImage(_ image: UIImage, id: String, sdf: Bool = false, stretchX: [ImageStretches] = [], stretchY: [ImageStretches] = [], content: ImageContent? = nil) throws {
+    public func addImage(_ image: UIImage,
+                         id: String,
+                         sdf: Bool = false,
+                         stretchX: [ImageStretches],
+                         stretchY: [ImageStretches],
+                         content: ImageContent?) throws {
```
Comment (macdrevx) [inline Sources/MapboxMaps/Style/Style.swift:479]: Let's retain the default value for `content`.

File: Sources/MapboxMaps/Location/LocationStyleProtocol.swift (hunk 1 above)
Comment (neelmistry94) [reply Sources/MapboxMaps/Location/LocationStyleProtocol.swift:35]: I have to dig into this further, I'm not sure how involved the `LocationStyleProtocol` is.

Comment (macdrevx) [reply Sources/MapboxMaps/Location/LocationStyleProtocol.swift:35]: I tried this locally — if you just delete this entire protocol extension, there just end up being 2 places that were depending on the default parameter for `layerPosition` assigned here where you'll need to explicitly pass nil instead. I think that's a pretty reasonable tradeoff for deleting these 20 lines of confusing code.

File: Sources/MapboxMaps/Location/LocationStyleProtocol.swift [hunk 3]
```diff
@@ -8,15 +8,30 @@ internal protocol LocationStyleProtocol: AnyObject {
...
+//                           stretchY: [ImageStretches],
+//                           content: ImageContent? = nil) throws {
+//        try addImage(image,
+//                     id: id,
+//                     sdf: sdf,
+//                     stretchX: stretchX,
+//                     stretchY: stretchY,
+//                     content: content)
+//    }
+//}
```
Comment (macdrevx) [inline Sources/MapboxMaps/Location/LocationStyleProtocol.swift:37]: Please delete instead of commenting-out

Comment (neelmistry94) [reply Sources/MapboxMaps/Location/LocationStyleProtocol.swift:37]: whoops, my bad

**Review by macdrevx — APPROVED** [review]

**Pull request merged**